### PR TITLE
[SPARK-31010][SQL][FOLLOW-UP] Deprecate untyped scala UDF

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?

Use scala annotation @deprecate to deprecate untyped scala UDF.

### Why are the changes needed?

After #27488, it's weird to see the untyped scala UDF will fail by default without deprecation.

### Does this PR introduce any user-facing change?

Yes, user will see the warning:
```
<console>:26: warning: method udf in object functions is deprecated (since 3.0.0): Untyped Scala UDF API is deprecated, please use typed Scala UDF API such as 'def udf[RT: TypeTag](f: Function0[RT]): UserDefinedFunction' instead.
       val myudf = udf(() => Math.random(), DoubleType)
                   ^
```

### How was this patch tested?

Tested manually.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -4732,6 +4732,8 @@ object functions {
    * @group udf_funcs
    * @since 2.0.0
    */
+  @deprecated("Untyped scala UDF API is deprecated, " +
+    "please use typed scala UDF API instead.", "3.0.0")
```
Comment (Ngone51) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: cc @cloud-fan

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: `scala` -> `Scala`

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: please use ... such as `def udf[RT: TypeTag](f: Function0[RT]): UserDefinedFunction` instead.

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Pull request closed without merging**